distributed/tests/test_client.py

```diff
@@ -3704,6 +3704,18 @@ def test_reconnect(loop):
     c.close()
 
 
+@gen_cluster(client=True, ncores=[], client_kwargs={"timeout": 0.5})
+def test_reconnect_timeout(c, s):
+    with captured_logger(logging.getLogger("distributed.client")) as logger:
+        yield s.close()
+        start = time()
+        while c.status != "closed":
+            yield c._update_scheduler_info()
```

Comment:
Maybe add a small sleep in here just so we don't overwhelm things (though probably it would be fine either way)

```suggestion
            yield c._update_scheduler_info()
            yield gen.sleep(0.05)
```